src/libraries/Microsoft.Extensions.Configuration.Json/tests/EmptyObjectTest.cs

```diff
@@ -0,0 +1,53 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Extensions.Configuration.Test;
+using Xunit;
+
+namespace Microsoft.Extensions.Configuration.Json.Test
+{
+    public class EmptyObjectTest
+    {
+        [Fact]
+        public void EmptyObject_AddsAsNull()
+        {
+            var json = @"{
+                ""key"": { },
+            }";
+
+            var jsonConfigSource = new JsonConfigurationProvider(new JsonConfigurationSource());
+            jsonConfigSource.Load(TestStreamHelpers.StringToStream(json));
+
+            Assert.Null(jsonConfigSource.Get("key"));
```

Comment:
Given that this returns `""` instead of `null` when `"key": null`, it seems weird for this to return `null` instead of `""`. This is especially true given the fact that #40827 was closed as by-design.

Out of curiosity, what does `jsonConfigSource.Get("non-existent-key")` return?

Reply:
I will double check tomorrow, but I believe it null. Also the value of a node which has children is null, if we set it to "", then it will have value "" even if we later add children from another provider.
I think there could be some other side effect of we set it to "", but I will have to check tomorrow.

Reply:
> Also the value of a node which has children is null

That's a very good point. While it's weird that explicitly setting a json property to `null` gives an empty string when inspected via the JsonConfigurationProvider, it would be more weird to have getting `null` for a non-empty section and `""` for an empty one. I'm fine with this as is then.